Destiny.ScrimTracker.Api/Controllers/GuardianController.cs

```diff
@@ -5,55 +5,90 @@
 using Destiny.ScrimTracker.Logic.Services;
+using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.Extensions.Logging;
 
 namespace Destiny.ScrimTracker.Api.Controllers
 {
     [ApiController]
     [Route("[controller]")]
-    public class GuardianController : ControllerBase
+    public class GuardianController : Controller
     {
+        private readonly IMatchService _matchService;
         private readonly IGuardianService _guardianService;
         private readonly ILogger<GuardianController> _logger;
 
-        public GuardianController(IGuardianService guardianService, ILogger<GuardianController> logger)
+        public GuardianController(IMatchService matchService, IGuardianService guardianService, ILogger<GuardianController> logger)
         {
+            _matchService = matchService;
             _guardianService = guardianService;
             _logger = logger;
         }
 
         [HttpGet]
-        public IEnumerable<Guardian> Get()
+        public IActionResult Get()
         {
-            return _guardianService.GetGuardians();
+            var guardianSnapshots = _guardianService.GetGuardians();
+            return View(guardianSnapshots);
         }
 
-        [HttpGet("{guardianId}")]
-        public Guardian GetGuardian(string guardianId)
+        [HttpGet("/add_guardian")]
+        [Authorize]
+        public IActionResult AddGuardianView()
         {
-            return _guardianService.GetGuardian(guardianId);
+            return View();
         }
-
-        [HttpPost]
-        public string Post(CreateGuardianRequest request)
+        
+        [HttpPost("/add_guardian")]
+        [Authorize]
+        public IActionResult AddGuardian([FromForm] CreateGuardianRequest request)
         {
+            if (!ModelState.IsValid)
+            {
+                return RedirectToAction("AddGuardianView");
+            }
+            
             var guardian = request.ToGuardian();
-            return _guardianService.CreateGuardian(guardian);
+            _guardianService.CreateGuardian(guardian);
+            return Redirect("/guardian");
+        }
+        
+        [HttpGet("{guardianId}")]
+        public IActionResult GetGuardian(string guardianId)
+        {
+            var guardian = _guardianService.GetGuardian(guardianId);
+            var guardianElo = _guardianService.GetGuardianElo(guardianId);
+            var guardianEfficiency = _guardianService.GetGuardianEfficiency(guardianId);
+            var matchCount = _matchService.GetMatchResultsForGuardian(guardianId).Count();
+            
+            var guardianHistory = new GuardianHistory()
+            {
+                Guardian = guardian,
+                EfficiencyHistory = guardianEfficiency,
+                EloHistory = guardianElo,
+                MatchCount = matchCount
+            };
+            
+            return View(guardianHistory);
         }
 
         [HttpPut("{guardianId}")]
+        [Authorize]

```

Comment:
Should this method redirect to the guardian that was updated?

Reply:
doneso